### PR TITLE
Proper handling of the isWebView config flag

### PR DESCRIPTION
**Pull request by atanasdb**

### CODE REVIEW

File: TDConnectIosSdk/OAuth2Module.swift [hunk 1]
```diff
@@ -191,17 +187,30 @@ open class OAuth2Module: NSObject, AuthzModule, SFSafariViewControllerDelegate {
...
+                return
+            }
+            if #available(iOS 9.0, *) {
+                browserType = .safariViewController;
+                let safariViewController = SFSafariViewController(url: url as URL)
+                safariViewController.delegate = self
+                UIApplication.shared.tdcTopViewController?.present(safariViewController, animated: true, completion: nil)
+                return
+            }
+            browserType = .safariExternalBrowser;
```
Comment (jorunfa) [inline TDConnectIosSdk/OAuth2Module.swift]: What will currently happen if we're on iOS 11 and we're not using a WebView? The code above is triggered and an external browser is used? I don't think there is a problem with that, I just want to make sure I understand.

File: TDConnectIosSdk/OAuth2Module.swift [hunk 2]
```diff
@@ -224,7 +233,7 @@ open class OAuth2Module: NSObject, AuthzModule, SFSafariViewControllerDelegate {
     }
     
     func callCompletion(success: AnyObject?, error: NSError?, completionHandler: @escaping (AnyObject?, NSError?) -> Void) {
-        if self.config.isWebView {
+        if browserType == .safariViewController {
```
Comment (jorunfa) [inline TDConnectIosSdk/OAuth2Module.swift]: Isn't this a reversal of how it used to work? That is, we used to trigger it when it was a WebView, now we trigger this if it is a `SFSafariViewController`?

Comment (atanasdb) [reply TDConnectIosSdk/OAuth2Module.swift]: You are right, has to be added for WebViews as well. But it also needs to stay for SFSafariViewController.
In the all code it was triggered for SFSafariViewController too. (it's just that SFSafariViewController was created even when webview was true for ios >= 9)

File: TDConnectIosSdk/OAuth2Module.swift (hunk 1 above)
Comment (atanasdb) [reply TDConnectIosSdk/OAuth2Module.swift]: no, there's a return there so safariAuthenticationSession will be used.
But I agree it's not very readable, will refactor it a bit.

File: TDConnectIosSdk/OAuth2Module.swift [hunk 3]
```diff
@@ -169,6 +181,9 @@ open class OAuth2Module: NSObject, AuthzModule, SFSafariViewControllerDelegate {
         // update state to 'Pending'
         self.state = .authorizationStatePendingExternalApproval
 
+        // get the user agent we will use for authentication
+        self.browserType = getBrowserTypeToUse();
```
Comment (jorunfa) [inline TDConnectIosSdk/OAuth2Module.swift]: Nice.

**Pull request merged**